### ISSUE TITLE
Should [Standard].Principle1.Guideline1_1.1_1_1.H67.2 be a notice?

### ISSUE THREAD

**Issue by 36degrees** [squizlabs/HTML_CodeSniffer#206]
*NOTE: This issue was originally [filed against pa11y](https://github.com/pa11y/pa11y/issues/372), but they asked us to record it here as it this is where their rules come from.*

Given a simple page with a single decorative image which can be safely ignored by assistive technologies:

```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>My Website</title>
  </head>

  <body>
    <img src="http://pa11y.org/resources/brand/logo.svg" alt="">
  </body>
</html>
```

When running the code sniffer against it, you get a warning for [Standard].Principle1.Guideline1_1.1_1_1.H67.2.

I think that this is being raised to prompt the user to manually verify whether this is the correct behaviour for the image as it depends on the image content?

However, when this sort of manual verification is prompted elsewhere it is raised as a `notice` with guidance to help the user verify that something is correct, such as with titles:

> Notice: Check that the title element describes the document.

I would suggest re-classifying WCAG2AA.Principle1.Guideline1_1.1_1_1.H67.2 as a notice, and modifying the text to something like:

> Notice: Img element is marked so that it is ignored by Assistive Technology. Check that this behaviour is correct.

We have seen users investing time and effect trying to 'fix' these warnings, despite the implementation being correct, because the way it is reported is confusing in suggesting something is 'definitely wrong'.

**Comment by ironikart** [comment]
Notifications are classified in HTMLCS as something that cannot be automatically tested (by this library), whereas warnings are something that can be tested but require human review. In this case I think a warning is the most appropriate classification as empty alternate text needs to be reviewed. An example is an image that contains text, we can't test for that so it requires a human to confirm whether an empty alt tag is appropriate.

**Comment by selfthinker** [comment]
This is not very consistent.

I've seen notifications that have been tested (for example, detecting if a layout table is present) but are not a warning.

And it's not just empty alternate text that needs to be reviewed, it's also non-empty alt text. There is a notice for reviewing non-empty alt text but for empty alt text it's a warning. Both can be detected, at least in the sense that it's either there or it isn't. And for neither can it be detected if it's correct or not (at least not by this library). It makes more sense to treat both cases the same way.

Apart from the inconsistency, I think there is one main issue with the way it is presented:
If you define what a 'warning' is in HTMLCS' case, it should either be automatically understood by its users (and I suspect it isn't because most users would understand it to mean that something is wrong and not that it should be reviewed) or it should be documented somewhere in a way which most users would read or at least find easily.
I searched but couldn't find any description of the issue types so far. Do you have that documented anywhere?

**Comment by gsherwood** [comment]
@selfthinker You might find this helpful as it shows the WCAG2 standard with each section marked error/warning/notice as HTMLCS sees it: http://squizlabs.github.io/HTML_CodeSniffer/Standards/WCAG2/

**Comment by selfthinker** [comment]
Thanks @gsherwood, that's very helpful!
Now that I tried to find where that is linked from I found it on the homepage. I tried to find something like that on the homepage before but I guess I didn't see the links initially because they are styled like buttons and I didn't expect them to lead anywhere.
